src/drivers/intel/ssp/ssp.c

```diff
@@ -205,6 +205,18 @@ static void ssp_bclk_disable_unprepare(struct dai *dai)
 	ssp->clk_active &= ~SSP_CLK_BCLK_ACTIVE;
 }
 
+static bool ssp_clks_active(struct ssp_pdata *ssp)
+{
+	if (ssp->state[SOF_IPC_STREAM_PLAYBACK] == COMP_STATE_ACTIVE ||
+	    ssp->state[SOF_IPC_STREAM_CAPTURE] == COMP_STATE_ACTIVE)
+		return true;
+
+	if (ssp->clk_active & (SSP_CLK_MCLK_ACTIVE | SSP_CLK_BCLK_ACTIVE))
+		return true;
```

Comment:
Sorry, I didn't get why we need to check both `ssp->clk_active` and `ssp->state`? Isn't checking `ssp->clk_active` enough?

Reply:
@bardliao That is probably unnecessary. I was just not usre whether "ssp->clk_active" is always updated. I.e. can clocks be enabled without setting this variable (if the early-clock mechanism is not used). I guess so, and in that case, L210-212 can be removed.

Reply:
Ack, @bardliao  you are right. The "clk_active" field is set in all code paths where BCLK or MCLK is enabled. So we can simplify the patch. New version uploaded.